src/main/java/org/openbase/bco/bcozy/model/UserData.java

```diff
@@ -0,0 +1,201 @@
+        firstname.setValue(getOrEmptyString(userConfig.getFirstName()));
+        lastName.setValue(getOrEmptyString(userConfig.getLastName()));
+        occupant.setValue(userConfig.getOccupant());
+        admin.setValue(CachedAuthenticationRemote.getRemote().isAdmin(userId.get()).get(2, TimeUnit.SECONDS));
+
+        if (userId.get() != null) {
+            groups.clear();
+            groups.addAll(AuthorizationGroups.getGroupsByUser(userId.get()));
+        }
+
+    }
+
+
+    private String getOrEmptyString(String string) {
+        return string != null ? string : "";
+    }
+
+    public boolean isUnsaved() {
+        return getUserId() == null || getUserId().isEmpty();
+    }
+
+    public String getUserId() {
+        return userId.get();
+    }
+
+    public boolean hasUserId() {
+        return userId.get() != null && !userId.get().isEmpty();
+    }
+
+    public String getPhone() {
+        return phone.get();
+    }
+
+    public StringProperty phoneProperty() {
+        return phone;
+    }
+
+    public String getUserName() {
+        return userName.get();
+    }
+
+    public StringProperty userNameProperty() {
+        return userName;
+    }
+
+    public String getMail() {
+        return mail.get();
+    }
+
+    public StringProperty mailProperty() {
+        return mail;
+    }
+
+    public String getFirstname() {
+        return firstname.get();
+    }
+
+    public StringProperty firstnameProperty() {
+        return firstname;
+    }
+
+    public String getLastName() {
+        return lastName.get();
+    }
+
+    public StringProperty lastNameProperty() {
+        return lastName;
+    }
+
+    public boolean isOccupant() {
+        return occupant.get();
+    }
+
+    public BooleanProperty occupantProperty() {
+        return occupant;
+    }
+
+    public UserConfigType.UserConfig getUserConfig() {
+        UserConfigType.UserConfig user = UserConfigType.UserConfig.newBuilder()
+                .setUserName(getOrEmptyString(userName.get()))
+                .setFirstName(getOrEmptyString(firstname.get()))
+                .setLastName(getOrEmptyString(lastName.get()))
+                .setEmail(getOrEmptyString(mail.get()))
+                .setMobilePhoneNumber(getOrEmptyString(phone.get()))
+                .setOccupant(occupant.get())
+                .build();
+
+        return user;
+    }
+
+    public boolean isAdmin() {
+        return admin.get();
+    }
+
+    public BooleanProperty adminProperty() {
+        return admin;
+    }
+
+    public List<UnitConfigType.UnitConfig> getGroups() {
+        return groups;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        UserData userData = (UserData) o;
+
+        return userId != null ? userId.equals(userData.userId) : userData.userId == null;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
This can do the HashCodeBuilder (Apache Commons Lang 3.6 API) for you:
https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/HashCodeBuilder.html

Reply:
Replaced with Objects.hashCode, which is even simpler, hope this is ok?